common_drivers/fru_tools/fru_tools.c

```diff
@@ -610,6 +610,7 @@ static unsigned int insert_str(unsigned char *buf, const unsigned char * str, bo
 			}
 			buf[0] = tmp | (FRU_STRING_ASCII << 6);
 			memcpy(&buf[1], &str[1], tmp);
+			free(six);
 			return tmp + 1;
 		}
```

Comment:
i think we may also need to free this after line 615, to make sure we don't leak in case `tmp >= 0`

Reply:
my bad;
i didn't see it at line 623